pkg/xds/topology/outbound.go

```diff
@@ -199,6 +199,10 @@ func buildExternalServiceEndpoint(externalService *mesh_core.ExternalServiceReso
 			mesh.GetMeta().GetName(), loader),
 		AllowRenegotiation: externalService.Spec.GetNetworking().GetTls().GetAllowRenegotiation().GetValue(),
 	}
+	if sni := externalService.Spec.GetNetworking().GetTls().GetSni(); sni != nil {
+		value := sni.GetValue()
+		es.SNI = &value
+	}
```

Comment:
```suggestion
        AllowRenegotiation: externalService.Spec.GetNetworking().GetTls().GetAllowRenegotiation().GetValue(),
        ServerName: externalService.Spec.GetNetworking().GetTls().GetServerName().GetValue(),
    }
```

Reply:
But we only want to use `tls.ServerName` if user defined `tls.ServerName`. Otherwise we want to use value generated by Kuma. That's why there is an if for `!= nil`

Reply:
The GetValue will be "" if the user hasn't set anything, which I think is what you want when you try to use it.

Reply:
Ok, here is what I want
* If ServerName is not set (== nil) then I want to use autogenerated ServerName by Kuma (which is just hostname of the destination)
* If ServerName is set to `""` then I want to send empty SNI
* If ServerName is set to `xyz` I want to override default autogenerated ServerName by Kuma

`GetValue()` will not handle the first case, because it will treat `nil` as `""`

Reply:
> If ServerName is set to "" then I want to send empty SNI

Why would anyone need to send an empty SNI?

Reply:
To be honest, I don't know any actual proxy that will require this, but I'd rather leave the possibility to set this to an empty string. If you feel strongly about this, I can remove this capability.

Reply:
IIUC you can't send an empty server name since it is defined as `opaque HostName<1..2^16-1>;`

https://www.rfc-editor.org/rfc/rfc6066#section-3

Reply:
ok, removed capability of setting empty SNI.